### PR TITLE
Optional untar on dbs

### LINKED ISSUE #196: Run untar on DBs of tools actually using
### Description of feature

Going through ToDos before first release. Shall we filter to only run UNTAR on DBs of tools actually using? Shall we make optional whether to save?

https://github.com/nf-core/taxprofiler/blob/578081df2e38b2de25509e683a13cd3d22ac772a/subworkflows/local/db_check.nf#L40-L50

### PR DESCRIPTION
**Pull request by sofstam**
Closes #196 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/db_check.nf [hunk 1]
```diff
@@ -2,14 +2,16 @@
 // Check input samplesheet and get read channels
 //
 
-include { UNTAR            } from '../../modules/nf-core/untar/main'
+include { UNTAR                  } from '../../modules/nf-core/untar/main'
```
Comment (jfy133) [inline subworkflows/local/db_check.nf:5]:
```suggestion
include { UNTAR } from '../../modules/nf-core/untar/main'
```

**Review by jfy133 — APPROVED** [review]
If it works in testing LGTM!

**Comment by Midnighter** [comment]
I probably would have opted for a little function. Something like (untested):

```nextflow
ch_input_untar = ch_dbs_for_untar
    .untar
    .filter { filter_active_tools(it[0]) }


def filter_active_tools(meta) {
    switch (meta.tool) {
        case 'bracken':
            return params.run_bracken
        case 'centrifuge':
            return params.run_centrifuge
        case 'kaiju':
            return params.run_kaiju
        case 'kraken2':
            return params.run_kraken2
        case 'krakenuniq':
            return params.run_krakenuniq
        case 'malt':
            return params.run_malt
        case 'metaphlan3':
            return params.run_metaphlan3
        default:
            error "Unknown taxonomic profiler '${meta.tool}'."
    }
}
```

**Comment by jfy133** [comment]
Might be nice it's certainly a bit cleaner, That said I sort of dislike functions in nextflow as they are often quite hidden... but I guess we could 'load' from a separate file via an `include`-like statement... maybe something for later :+1:

**Pull request merged**